File: 369/HexColor.cpp
#include <iostream>
#include <sstream>
std::string decimalToHex(int decimal);
std::string hexcolor(int red, int green, int blue);
int main()
{
   std::cout<<hexcolor(255, 99, 71) << std::endl;
   std::cout<<hexcolor(0, 0, 205) << std::endl;
}

std::string decimalToHex(int decimal)
{
  std::stringstream ss;
  ss << std::hex << decimal;
  std::string hex(ss.str());  
  // handle preceding zeros
  if(hex.length() == 1) hex.append("0",0,1);
  return hex;
}
std::string hexcolor(int red, int green, int blue)
{
   std::string hexColor = "#";
   hexColor.append(decimalToHex(red));
   hexColor.append(decimalToHex(green));
   hexColor.append(decimalToHex(blue));
   return hexColor;
}

File: 369/HexColor.java
public class HexColor{
	public static void main(String [] args){
		
		System.out.println(hexcolor(255, 99, 71));
		System.out.println(hexcolor(184, 134, 11));
		System.out.println(hexcolor(189, 183, 107));
		System.out.println(hexcolor(0, 0, 205));
		
	}

	private static String hexcolor(int red, int green, int blue){
		StringBuilder hexColor = new StringBuilder("#");
		hexColor.append(decimalToHex(red));
		hexColor.append(decimalToHex(green));
		hexColor.append(decimalToHex(blue));
		return hexColor.toString();
	}

	private static String decimalToHex(int decimal){
		String hex = String.format("%X", decimal);
		// handle preceding zeros
		if(hex.length() == 1) hex = "0".concat(hex);
		return hex;
	}
	
}
